packages/types/src/interfaces/extrinsic/v1/Extrinsic.ts

```diff
@@ -0,0 +1,89 @@
+import { GenericExtrinsic, UInt } from '@polkadot/types';
+import { SignatureOptions } from '@cennznet/types/interfaces/extrinsic/types';
+import { AnyNumber, AnyU8a, ExtrinsicPayloadValue, IKeyringPair, Registry } from '@polkadot/types/types';
+import { Address, Call } from '@polkadot/types/interfaces/runtime';
+import { ExtrinsicValueV4 } from '@polkadot/types/extrinsic/v4/Extrinsic';
+import { Api } from '@cennznet/api';
+import { PaymentOptions } from '@cennznet/api/derives/types';
+
+export default class CENNZnetExtrinsic extends GenericExtrinsic{
+  private signaturePayloadOptions: SignatureOptions | ExtrinsicPayloadValue;
+
+  constructor(registry: Registry, value?: GenericExtrinsic | ExtrinsicValueV4 | AnyU8a | Call) {
+    super(registry, value);
+    this.signaturePayloadOptions = {
+      blockHash: undefined,
+      era: undefined,
+      genesisHash: undefined,
+      method: undefined,
+      specVersion: undefined,
+      nonce: undefined,
+      runtimeVersion: undefined,
+      signedExtensions: undefined,
+      signer: undefined,
+      tip: undefined,
+      transactionPayment: undefined,
```

Comment:
export interface SignatureOptions {
    blockHash: Uint8Array | string;
    era?: IExtrinsicEra;
    genesisHash: Uint8Array | string;
    nonce: AnyNumber;
    runtimeVersion: IRuntimeVersion;
    signedExtensions?: string[];
    signer?: Signer;
    tip?: AnyNumber;
}

era, signedExtension, signer, tip and transactionPayment are optional field and its ok even if we don't explicit set them to undefined

Reply:
yup sweet can update that